airbyte-config/config-persistence/src/main/java/io/airbyte/config/persistence/ConfigRepository.java

```diff
@@ -794,6 +827,68 @@ private Map<UUID, AirbyteCatalog> findCatalogByHash(final String catalogHash, fi
     return result;
   }
 
+  public List<SourceConnection> getSourceConnections(final List<UUID> sourceIds) throws IOException {
+    return database.query(ctx -> getSourceConnections(sourceIds, ctx));
+  }
+
+  public List<SourceConnection> getSourceConnections(final List<UUID> sourceIds, final DSLContext ctx) {
+    final Result<Record> records = ctx
+        .select(asterisk())
+        .from(ACTOR)
+        .where(ACTOR.ACTOR_TYPE.eq(ActorType.source), ACTOR.ID.in(sourceIds))
+        .fetch();
+
+    return records.stream().map(DbConverter::buildSourceConnection).toList();
+  }
+
```

Comment:
some thoughts:
- most of the following functions seems to only be used in the webbackend list handler.
- do we foresee using them in other places? If we don't, what about combining some of these functions together? e.g. combining `getSourceConnections` and `getSourceDefinitionsFromSourceIds` into one query + function. The single query should make it clearer what the query is for, and minimise database query roundtrips.
- I feel like the small number of functions will make it clear what some of these are used for and improve readability.

Reply:
@davinchia I took a shot at this, you can see my approach in this commit: https://github.com/airbytehq/airbyte/pull/17360/commits/cc22381e26e844c5bc487057e3d41caead01a237

Basically, I introduced a new `record` type that holds the combined source/definition result from the single query, which the API layer can then process into a `SourceRead` with ease. Let me know what you think!

Reply:
Very nice!

Reply:
I like the use of the record class so it's clear in types what this is for.